src/core/Akka.Remote/Transport/Streaming/SslStreamTransport.cs

```diff
@@ -0,0 +1,162 @@
+
+            bool allowUntrusted = config.GetBoolean("insecure-allow-untrusted-server-certificate");
+            if (allowUntrusted)
+            {
+                CheckServerCertificateRevocation = false;
+                CertificateValidationCallback = (sender, certificate, chain, errors) => true;
+            }
+            else
+            {
+                CheckServerCertificateRevocation = config.GetBoolean("check-server-certificate-revocation", true);
+                CertificateValidationCallback = null;
+            }
+
+
+            StoreLocation location;
+            string storeLocationParameter = "store-location";
+            string storeLocationString = config.GetString(storeLocationParameter);
+            if (!Enum.TryParse(storeLocationString, out location))
+                throw new ArgumentException($"Invalid StoreLocation '{storeLocationString}'", storeLocationParameter);
+
+            //TODO Get the server certificate from thumbprint or subject name
+        }
+
+        public static X509Certificate2 GetCertificateFromThumbprint(StoreLocation location, string thumbprint)
+        {
+            if (thumbprint == null)
+                return null;
+
+            // When copy pasting from the Certificate UI, it often start with the Left-to-right mark
+            // Remove it or we won't find the certificate.
+            if (thumbprint[0] == 0x200E)
+                thumbprint = thumbprint.Substring(1);
+
+            X509Store store = new X509Store(StoreName.My, location);
+
+            store.Open(OpenFlags.ReadOnly);
+
+            var certificates = store.Certificates.Find(X509FindType.FindByThumbprint, thumbprint, false);
+
+            X509Certificate2 result = null;
+
+            if (certificates.Count > 0)
+                result = certificates[0];
+
+            store.Close();
+
+            return result;
+        }
+
+        public static X509Certificate2 GetCertificateFromSubjectName(StoreLocation location, string subjectName)
+        {
+            X509Store store = new X509Store(StoreName.My, location);
+
+            store.Open(OpenFlags.ReadOnly);
+
+            var certificates = store.Certificates.Find(X509FindType.FindBySubjectName, subjectName, false);
+
+            X509Certificate2 result = null;
+
+            if (certificates.Count == 1)
+            {
+                result = certificates[0];
+            }
+            else if (certificates.Count > 1)
+            {
+                result = certificates.Cast<X509Certificate2>()
+                                     .OrderByDescending(item => item.NotAfter)
+                                     .First();
+            }
+
+            store.Close();
+
+            return result;
+        }
+    }
+
+    public class SslStreamTransport : NetworkStreamTransport
+    {
+        protected new SslStreamTransportSettings Settings { get; }
+
+        public override string SchemeIdentifier
+        {
+            get { return "ssl"; }
```

Comment:
Should be `ssl.tcp` or whatever the underlying transport is - see equivalent JVM code: https://github.com/akka/akka/blob/master/akka-remote/src/main/scala/akka/remote/transport/netty/NettyTransport.scala#L272